### DISCUSSION TITLE
Best Practices for a component edit modal?  Should I pass the full Model?  The property I'm editing?

### DISCUSSION THREAD

**Discussion by mendaloth** [livewire/livewire#4345] [1. Help] [answered]
I have Contacts which have a bunch of Emails.  Right now when opening the modal window to edit them I just call the component this way:
                    <livewire:edit-emails-modal :contact="$contact"/>

It works, but am I creating unneeded overhead?  If I wanted to just pass the collection of emails could I do <livewire:edit-emails-modal :contact="$contact->emails"/>, what would the public property type be?  I'm also do this for the edit modal to be able to ADD emails, without access to the $contact email would it even be possible to add emails?

**Comment by joshhanley** [comment, marked as answer]
@mendaloth I personally would just pass the contact model through like if your first example 🙂 that way you have access to the contact model for adding new emails like you suggested.

Hope this helps!
